src/App.tsx

```diff
@@ -1,14 +1,89 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useEffect,
+  useState,
+  useCallback,
+  useMemo,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { ErrorMessage } from './components/ErrorMessage';
+
+import { Todo } from './types/Todo';
+import { TodoWithUser } from './types/TodoWithUser';
+import { TodoCompletionFilter } from './types/TodoCompletionFilter';
+
+import { getTodos, getUser } from './api';
+
+import { filterTodos } from './helpers';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isTodosLoading, setIsTodosLoading] = useState(false);
+  const [hasTodosLoadingError, setHasTodosLoadingError] = useState(false);
+  const [isTodosLoadInitialized, setIsTodosLoadInitialized] = useState(false);
```

Comment:
do you need it? you already have `isTodosLoading`

Reply:
Yes, absolutely need it for handling empty response, e.g. empty array of todos. There is no way to correctly handle this situation playing around only with isLoading, hasError and todos.length === 0.

I have received this comment twice now :) Simply nobody has implemented this feature I guess and this situation (correct but empty response) is not detectable through a GitHub review.

![image](https://user-images.githubusercontent.com/50755505/231442022-3e727d2c-fbca-427a-a1e8-265d7dbe7b4b.png)
![image](https://user-images.githubusercontent.com/50755505/231442821-16039a6f-6c70-436e-bff0-79ba7145d3a5.png)